3scaleAdapter/pkg/threescale/proxy_conf.go

```diff
@@ -251,6 +269,33 @@ func (pc *ProxyConfigCache) shouldRefresh(currentTime time.Time, expiryTime time
 	return currentTime.Add(pc.refreshBuffer).After(expiryTime)
 }
 
+func (pc *ProxyConfigCache) addMisbehavingHost(url string, err error) {
+	var misbehaving bool
+
+	switch errT := err.(type) {
+	case net.Error:
+		if errT.Timeout() {
+			misbehaving = true
+		}
+	case sysC.ApiErr:
+		if errT.Code() >= 500 && errT.Code() < 600 {
+			misbehaving = true
+		}
+	}
+	if misbehaving {
+		pc.misbehavingHosts = append(pc.misbehavingHosts, url)
```

Comment:
we need to extract the host, as the URL might change but the host remains the same